src/libraries/Common/src/System/Buffers/ArrayBufferWriter.cs

```diff
@@ -15,10 +15,14 @@ namespace System.Buffers
 #endif
     sealed class ArrayBufferWriter<T> : IBufferWriter<T>
     {
+        // Copy of Array.MaxArrayLength. For byte arrays the limit is slightly larger
+        private const int MaxArrayLength = 0X7FEFFFFF;
```

Comment:
Can we just use `Array.MaxArrayLength`, so we don't have this constant in multiple places?

Reply:
This file is compiled in NS2.0 assemblies, so it needs to have a local copy. I agree it would be nice to have an API for this - reactivating https://github.com/dotnet/runtime/issues/31366

Reply:
Will it matter on full framework where MaxArrayLength might be smaller (I believe not due to how the buffer grows, but I thought I'd double check)?

Reply:
Right, it is not easy to find the maximum array length in .NET Framework. It depends on `gcAllowVeryLargeObjects` configuration setting. It just means that this code may throw OOM on .NET Framework even when it could have been avoided. I think that is fine.